File: dotnet-adapters-shortest-paths-core/main/java/com/programmerare/shortestpaths/core/pathfactories/PathFactoryDefault.java
/*
* Copyright (c) Tomas Johansson , http://www.programmerare.com
* The code is made available under the terms of the MIT License.
* https://github.com/TomasJohansson/adapters-shortest-paths/blob/master/adapters-shortest-paths-core/License.txt
*/
package com.programmerare.shortestpaths.core.pathfactories;

import java.util.List;

import com.programmerare.shortestpaths.core.api.Edge;
import com.programmerare.shortestpaths.core.api.Path;
import com.programmerare.shortestpaths.core.api.Vertex;
import com.programmerare.shortestpaths.core.api.Weight;
import com.programmerare.shortestpaths.core.impl.PathImpl;

public final class PathFactoryDefault
	implements PathFactory<Path , Edge , Vertex , Weight>
{
	public Path createPath(final Weight totalWeight, final List<Edge> edges) {
		final Path pathDefault = PathImpl.createPath((Weight)totalWeight, edges);
		// System.out.println("PathFactory created PathFactoryDefault " + pathDefault.getClass());
		return pathDefault;
	}
}

File: dotnet-adapters-shortest-paths-core/src/com.programmerare.shortestpaths/core/pathfactories/PathFactoryDefault.cs
/*
* Copyright (c) Tomas Johansson , http://www.programmerare.com
* The code is made available under the terms of the MIT License.
* https://github.com/TomasJohansson/adapters-shortest-paths/blob/master/adapters-shortest-paths-core/License.txt
*/
using com.programmerare.shortestpaths.core.api;
using com.programmerare.shortestpaths.core.impl;
using System.Collections.Generic;

namespace com.programmerare.shortestpaths.core.pathfactories
{
    public sealed class PathFactoryDefault
	    : PathFactory<Path , Edge , Vertex , Weight>
    {
	    public Path CreatePath(Weight totalWeight, IList<Edge> edges) {
		    Path pathDefault = PathImpl.CreatePath((Weight)totalWeight, edges);
		    return pathDefault;
	    }
    }
}